File: VB/Test/WinSolution.Module/DomainObject1.vb
Imports Microsoft.VisualBasic
Imports System

Imports DevExpress.Xpo

Imports DevExpress.ExpressApp
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Persistent.Validation

Namespace WinSolution.Module
	<DefaultClassOptions> _
	Public Class DomainObject1
		Inherits BaseObject
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private _Name As String
		Public Property Name() As String
			Get
				Return _Name
			End Get
			Set(ByVal value As String)
				SetPropertyValue("Name", _Name, value)
			End Set
		End Property
		Private _DomainObject2 As DomainObject2
		<ImmediatePostData> _
		Public Property DomainObject2() As DomainObject2
			Get
				Return _DomainObject2
			End Get
			Set(ByVal value As DomainObject2)
				SetPropertyValue("DomainObject2", _DomainObject2, value)
			End Set
		End Property
	End Class

End Namespace

File: CS/Test/WinSolution.Module/DomainObject1.cs
using System;

using DevExpress.Xpo;

using DevExpress.ExpressApp;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Persistent.Validation;

namespace WinSolution.Module {
    [DefaultClassOptions]
    public class DomainObject1 : BaseObject {
        public DomainObject1(Session session) : base(session) { }
        private string _Name;
        public string Name {
            get { return _Name; }
            set { SetPropertyValue("Name", ref _Name, value); }
        }
        private DomainObject2 _DomainObject2;
        [ImmediatePostData]
        public DomainObject2 DomainObject2 {
            get { return _DomainObject2; }
            set { SetPropertyValue("DomainObject2", ref _DomainObject2, value); }
        }
    }

}
